pan3d/explorers/slice_explorer.py

```diff
@@ -0,0 +1,441 @@
+        s_lut.SetHueRange(0.666, 0.0)
+        s_lut.SetSaturationRange(1.0, 1.0)
+        s_lut.SetValueRange(1.0, 1.0)
+        d_lut.SetHueRange(0.666, 0.0)
+        d_lut.SetSaturationRange(1.0, 1.0)
+        d_lut.SetValueRange(1.0, 1.0)
+    elif preset == "Inv Rainbow":
+        s_lut.SetHueRange(0.0, 0.666)
+        s_lut.SetSaturationRange(1.0, 1.0)
+        s_lut.SetValueRange(1.0, 1.0)
+        d_lut.SetHueRange(0.0, 0.666)
+        d_lut.SetSaturationRange(1.0, 1.0)
+        d_lut.SetValueRange(1.0, 1.0)
+    elif preset == "Greyscale":
+        s_lut.SetHueRange(0.0, 0.0)
+        s_lut.SetSaturationRange(0.0, 0.0)
+        s_lut.SetValueRange(0.0, 1.0)
+        d_lut.SetHueRange(0.0, 0.0)
+        d_lut.SetSaturationRange(0.0, 0.0)
+        d_lut.SetValueRange(0.0, 1.0)
+    elif preset == "Inv Greyscale":
+        s_lut.SetHueRange(0.0, 0.666)
+        s_lut.SetSaturationRange(0.0, 0.0)
+        s_lut.SetValueRange(1.0, 0.0)
+        d_lut.SetHueRange(0.0, 0.666)
+        d_lut.SetSaturationRange(0.0, 0.0)
+        d_lut.SetValueRange(1.0, 0.0)
+    s_lut.Build()
+    d_lut.Build()
+
+def update_preset(actor: vtk.vtkActor, logcale : bool) -> None:
+    lut = actor.GetMapper().GetLookupTable()
+    if logcale:
+        print("Setting log scale")
+        lut.SetScaleToLog10()
+    else:
+        print("Setting linear scale")
+        lut.SetScaleToLinear()
+    lut.Build()
+
+@TrameApp()
+class SliceExplorer:
+    def __init__(self, builder : DatasetBuilder, server=None):
+        self.server = get_server(server)
+        self.builder = builder
+        self.dims    = dict([(x, y) for (x, y) in zip([builder.x, builder.y, builder.z], ['x', 'y','z']) if not x is None])
+        self.extents = builder.extents
+
+        self._ui = None
+        self.state.slice_dim = list(self.dims.keys())[0]
+        ext = self.extents[self.state.slice_dim]
+        self.state.dimval = float(ext[0] + (ext[1] - ext[0]) / 2)
+        self.state.dimmin = float(ext[0])
+        self.state.dimmax = float(ext[1])
+        self.state.varmin = 0.
+        self.state.varmax = 0.
+
+        self.t_cache = {}
+        self.vars = list(builder.dataset.data_vars.keys())
+        self.var_ranges = builder.var_ranges
+
+        self.render_window  = vtk.vtkRenderWindow()
+        self.renderer       = vtk.vtkRenderer()
+        self.interactor     = vtk.vtkRenderWindowInteractor()
+
+        extents = list(self.extents.values())
+        self.origin = [
+            float(extents[0][0] + (extents[0][1] - extents[0][0]) / 2), 
+            float(extents[1][0] + (extents[1][1] - extents[1][0]) / 2), 
+            float(extents[2][0] + (extents[2][1] - extents[2][0]) / 2)
+        ] 
+        self.normal = [1, 0, 0]
+        self.lengths = [
+            float(extents[0][1] - extents[0][0]),
+            float(extents[1][1] - extents[1][0]),
+            float(extents[2][1] - extents[2][0]),
+        ]
+
+        self.plane = vtk.vtkPlane()
```

Comment:
I would gather the vtk code inside `self._setup_viz()` and just call it in constructor